### PR TITLE
Ability to see if an action has attachments on list views

### PR DESCRIPTION
**Pull request by nmajor25**
Shortcut Story ID: [sc-32259]

If an action has one or more attachments, display a paperclip icon in list views that indicates it has attachments.

Icon was added to action list items on the following pages:

1. Worklist
2. Patient Dashboard
3. Patient Archive
4. Patient flow

Screen Shot 2022-11-18 at 3 36 08 PM

### CODE REVIEW

File: src/js/views/patients/worklist/worklist-list.scss [hunk 1]
```diff
@@ -28,7 +28,13 @@
...
   margin-right: 16px;
   vertical-align: middle;
-  width: 16px;
+}
+
+.worklist-list__attachment {
+  color: #999;
+  display: inline-block;
+  margin-left: 8px;
+  vertical-align: middle;
```
Comment (nmajor25) [inline src/js/views/patients/worklist/worklist-list.scss:29]: This `.worklist-list__attachment` and `.worklist-list__details` above it have almost identical properties. 

Maybe we could combine them and make use of `.u-margin--l-8` and `.u-margin--r-16` for the `margin-right`/`margin-left` differences?

File: test/integration/patients/patient/patient-flow.js [hunk 2]
```diff
@@ -814,6 +817,8 @@ context('patient flow page', function() {
         return fx;
       })
       .routeActionActivity()
+      .routePatientField()
+      .routeActionComments()
```
Comment (nmajor25) [inline test/integration/patients/patient/patient-flow.js:821]: These are unrelated to the topic of this PR, but were causing some 500 errors in the tests. Noticed a few more that I added throughout this PR.

File: src/scss/modules/table-list.scss [hunk 3]
```diff
@@ -50,6 +50,10 @@
 
 .table-list__meta {
   margin-right: 8px;
+
+  &:empty {
+    margin-right: 0;
+  }
```
Comment (nmajor25) [inline src/scss/modules/table-list.scss]: There are times when the form button region (`[data-form-region]`) is empty but still has `margin-right: 8px` applied to it. That sometimes adds unnecessary extra margin when the attachments icon is being shown.

This stops that from happening.

File: src/js/views/patients/worklist/worklist-list.scss (hunk 1 above)
Comment (paulfalgout) [reply src/js/views/patients/worklist/worklist-list.scss:29]: yep I think that's ideal.  perhaps the class can be something more generic.  `__display-icon` or something?

File: src/scss/modules/table-list.scss (hunk 3 above)
Comment (paulfalgout) [reply src/scss/modules/table-list.scss]: could also be `.table-list__meta:not(:empty)`

File: test/integration/patients/patient/patient-flow.js [hunk 4]
```diff
@@ -209,6 +210,7 @@ context('patient flow page', function() {
         expect($action.find('[data-owner-region]')).to.contain('NUR');
         expect($action.find('[data-due-day-region] .is-overdue')).to.exist;
         expect($action.find('[data-form-region]')).not.to.be.empty;
+        expect($action.find('.patient__action-attachment')).to.exist;
```
Comment (paulfalgout) [inline test/integration/patients/patient/patient-flow.js]: I think we should test for the icon directly over the style class

**Review by paulfalgout — CHANGES_REQUESTED** [review]

File: src/js/views/patients/worklist/worklist-list.scss (hunk 1 above)
Comment (nmajor25) [reply src/js/views/patients/worklist/worklist-list.scss:29]: That works well. Updated in latest push.

File: src/scss/modules/table-list.scss (hunk 3 above)
Comment (nmajor25) [reply src/scss/modules/table-list.scss]: Yup... definitely like that better. Updated in last push.

File: test/integration/patients/patient/patient-flow.js (hunk 4 above)
Comment (nmajor25) [reply test/integration/patients/patient/patient-flow.js]: Updated in last push.

**Review by paulfalgout — APPROVED** [review]

**Pull request merged**